hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/processor/SourceProcessors.java

```diff
@@ -336,16 +336,17 @@ public static ProcessorMetaSupplier streamSocketP(
 
     /**
      * Returns a supplier of processors for
-     * {@link Sources#fileWatcher(String, Charset, String)}.
+     * {@link Sources#fileWatcher(String, Charset, String, boolean)}.
      */
     @Nonnull
     public static ProcessorMetaSupplier streamFilesP(
             @Nonnull String watchedDirectory,
             @Nonnull Charset charset,
             @Nonnull String glob,
-            @Nonnull DistributedBiFunction<String, String, ?> mapOutputFn
+            @Nonnull DistributedBiFunction<String, String, ?> mapOutputFn,
+            boolean sharedFileSystem
```

Comment:
shouldn't be last param.
What about `Sources.files`?

Reply:
I've already added shared files system support to `Sources.files` with Avro PR
https://github.com/hazelcast/hazelcast-jet/pull/890/files#diff-ebaff9dcd50ec0708250477d8b796454R771
it was the last parameter there too

Reply:
ok, can we make it so that it's not after the lambda for both? code looks generally nicer if the lambda is the last parameter.